commit 7993bd430e84c695cc120c0e67ffb40a086aedea
Author: Luke Melia <luke@lukemelia.com>
Date:   Sat Apr 18 01:41:27 2009 -0400

    Avoid loading ActiveRecord::SessionStore on initialization if it is not in use

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index a04405a..6aa7489 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -420,7 +420,7 @@ Run `rake gems:install` to install the missing gems.
 
     def initialize_database_middleware
       if configuration.frameworks.include?(:active_record)
-        if ActionController::Base.session_store == ActiveRecord::SessionStore
+        if ActionController::Base.session_store.name == 'ActiveRecord::SessionStore'
           configuration.middleware.insert_before :"ActiveRecord::SessionStore", ActiveRecord::ConnectionAdapters::ConnectionManagement
           configuration.middleware.insert_before :"ActiveRecord::SessionStore", ActiveRecord::QueryCache
         else
